lib/kubeclient/common.rb

```diff
@@ -150,13 +150,19 @@ def self.parse_definition(kind, name)
         # But how?  If it differs from kind.downcase, kind's word boundaries don't apply.
         singular_name = kind.downcase
 
-        if name.start_with?(kind.downcase)
+        if name.start_with?(singular_name)
           plural_suffix = name[kind.downcase.length..-1]             # "es"
           singular_underscores = ClientMixin.underscore_entity(kind) # "component_status"
           method_names = [singular_underscores, singular_underscores + plural_suffix]
         else
-          # Something weird, can't infer underscores for plural so just give them up
-          method_names = [singular_name, name]
+          stripped_name = name.tr('-', '').tr('_', '') # clear separators
+          if stripped_name.start_with?(singular_name)
```

Comment:
Did you mean the opposite ? Since `singular_underscores` is shorten than `name` and fails for all tests.

```
name.tr('-', '_').start_with?(singular_underscores)
```

However, with this change, your suggested test case
`      SameUptoWordboundary sameup-toword-boundarys same_upto_wordboundary sameup_toword_boundarys
` fails.

This test (and other) passes without changing the proposed code.

Reply:
Sorry, yes, opposite.

The test "case" was not really a suggested case to add as-is, just example of `stripped_name.start_with?(singular_name)` behavior, which I think is not good:
```
get_same_upto_wordboundary 
get_sameup_toword_boundarys
```
And I think desired behavior here is the last fallback of dropping the dashes:
```
get_sameuptowordboundary 
get_sameuptowordboundarys
```

Reply:
Updated the PR to produce the desired naming. Add test with proper expectation for it.